modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java

```diff
@@ -2557,8 +2557,12 @@ public int execute(List<String> rawArgs) {
 
                         tryConnectMaxCount--;
                     }
-                    else
+                    else {
+                        if (tryConnectMaxCount == 0)
+                            throw new GridClientException("Authentication error, max retries exceeded");
```

Comment:
It is better to use more specific `GridClientAuthenticationException` here.

Reply:
Fixed